src/views/examples/Register.jsx

```diff
@@ -17,71 +18,209 @@ import {
-  }
 
-  passwordMatcher = e => {
-    this.setState({ confirmPassword: e.target.value });
-    if (this.state.password === e.target.value) {
-      this.setState({
-        passMatch: {
-          text: "Senhas conferem!",
-          status: 1
-        }
-      });
-    } else {
-      this.setState({
-        passMatch: {
-          text: "Senhas não conferem!",
-          status: 2
-        }
-      });
+    this.setState({ firstNameValid, errorMsg }, this.validateForm);
+  };
+
+  updateSecondName = secondName => {
+    this.setState({ secondName }, this.validateSecondName);
+  };
+
+  validateSecondName = () => {
+    const { secondName } = this.state;
+    let secondNameValid = true;
+    let errorMsg = { ...this.state.errorMsg };
+
+    // can't be blank/empty
+
+    if (secondName.length === 0) {
+      secondNameValid = false;
+      errorMsg.secondName = "Sobrenome não pode ser vazio!";
     }
+
+    this.setState({ secondNameValid, errorMsg }, this.validateForm);
   };
 
-  handleSubmit = () => {
-    // Form validation
-    // API Call
+  updateCPF = cpf => {
+    this.setState({ cpf }, this.validateCPF);
   };
 
-  showHide(e) {
-    e.preventDefault();
-    e.stopPropagation();
+  validateCPF = () => {
+    const { cpf } = this.state;
+    let cpfValid = true;
+    let errorMsg = { ...this.state.errorMsg };
+
+    if (cpfMask(cpf).length > 0 && cpfMask(cpf).length < 14) {
+      cpfValid = false;
+      errorMsg.cpf = "Insira um CPF válido!";
+    } else if (cpfMask(cpf).length === 0) {
+      cpfValid = false;
+      errorMsg.cpf = "CPF não pode ser vazio!";
+    }
+
+    this.setState({ cpfValid, errorMsg }, this.validateForm);
+  };
+
+  updateMatriculation = matriculation => {
+    this.setState({ matriculation }, this.validateMatriculation);
+  };
+
+  validateMatriculation = () => {
+    const { matriculation } = this.state;
+    let matriculationValid = true;
+    let errorMsg = { ...this.state.errorMsg };
+
+    if (matriculation.length === 0) {
+      matriculationValid = false;
+      errorMsg.matriculation = "Matrícula não pode ser vazia!";
+    }
+
+    this.setState({ matriculationValid, errorMsg }, this.validateForm);
+  };
+
+  updateEmail = email => {
+    this.setState({ email }, this.validateEmail);
+  };
+
+  validateEmail = () => {
+    const { email } = this.state;
+    let emailValid = true;
+    let errorMsg = { ...this.state.errorMsg };
+
+    // checks for format _@_._
+    if (!/^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email)) {
+      emailValid = false;
+      errorMsg.email = "Insira um e-mail válido!";
+    }
+
+    this.setState({ emailValid, errorMsg }, this.validateForm);
+  };
+
+  updatePassword = password => {
+    this.setState({ password }, this.validatePassword);
+  };
+
+  validatePassword = () => {
+    const { password, passwordConfirm } = this.state;
+    let passwordValid = true;
+    let passwordConfirmValid = true;
+    let errorMsg = { ...this.state.errorMsg };
+
+    // must be 6 characters
```

Comment:
Set this in the state configuration, latter if we need to change this, it'll be hard to find (after a time without see the code)

Reply:
Nope. Source [here.](https://medium.com/trainingcenter/mascara-de-cpf-com-react-javascript-a07719345c93)

Reply:
> Set this in the state configuration, latter if we need to change this, it'll be hard to find (after a time without see the code)

Can you give me an example? :thinking:

Reply:
In the env file:

```
REACT_APP_PASSWORD_LENGTH = 6
```

In the script file:

```javascript
import { config } from 'dotenv';

config();

class User extends Component {
  constructor(props) {
    super(props);

    this.state = { passwordLength: process.env.PASSWORD_LENGTH };
  }

  checkForm = (event) => {
    if(this.state.passwordLength)
        // ...
  }
}
```

Obs ¹: Its better to place the `config()` in the *index.js* or *app.js*, because the environment variables are inherited by all the components rendered in one of these files.

Obs ²: You could also use the `process.env.PASSWORD_LENGTH` inside the `checkForm()`, but if latter you need to change the key name in the env file, it will be a mass to change in the components that use this key, so using this approach, you just change one line and internalize what the component consumes